libraries/freertos_plus/standard/freertos_plus_tcp/source/portable/NetworkInterface/STM32Fxx/NetworkInterface.c

```diff
@@ -801,20 +814,24 @@ const ProtocolPacket_t *pxProtPacket = ( const ProtocolPacket_t * )pcBuffer;
 
 		if( pxIPHeader->ucProtocol == ipPROTOCOL_UDP )
 		{
-			uint16_t port = pxProtPacket->xUDPPacket.xUDPHeader.usDestinationPort;
+		uint16_t usSourcePort = FreeRTOS_ntohs( pxProtPacket->xUDPPacket.xUDPHeader.usSourcePort );
+		uint16_t usDestinationPort = FreeRTOS_ntohs( pxProtPacket->xUDPPacket.xUDPHeader.usDestinationPort );
 
-			if( ( xPortHasUDPSocket( port ) == pdFALSE )
+			if( ( xPortHasUDPSocket( pxProtPacket->xUDPPacket.xUDPHeader.usDestinationPort ) == pdFALSE )
 			#if ipconfigUSE_LLMNR == 1
-				&& ( port != FreeRTOS_ntohs( ipLLMNR_PORT ) )
+				&& ( usDestinationPort != ipLLMNR_PORT )
+				&& ( usSourcePort != ipLLMNR_PORT )
 			#endif
 			#if ipconfigUSE_NBNS == 1
-				&& ( port != FreeRTOS_ntohs( ipNBNS_PORT ) )
+				&& ( usDestinationPort != ipNBNS_PORT )
+				&& ( usSourcePort != ipNBNS_PORT )
 			#endif
 			#if ipconfigUSE_DNS == 1
-				&& ( pxProtPacket->xUDPPacket.xUDPHeader.usSourcePort != FreeRTOS_ntohs( ipDNS_PORT ) )
+				&& ( usSourcePort != ipDNS_PORT )
 			#endif
 				) {
 				/* Drop this packet, not for this device. */
+				/* FreeRTOS_printf( ( "Drop: UDP port %d -> %d\n", usSourcePort, usDestinationPort ) ); */
```

Comment:
I'd delete this debugging comment, or (better) add configurable debug logging to the driver.

Reply:
I always have this logging enabled. It can very useful when a user sees why his driver "doesn't seem to work": filtering of packets.
As you know, the macro `FreeRTOS_printf()` is already configurable and will be disabled in production..